src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.jsx

```diff
@@ -29,24 +34,105 @@ const HeatmapGroupBySettings = () => {
     );
   };
 
+  const getCellSetsOrder = () => {
+    const allCellSetsGroupBys = getCellSets(['cellSets', 'metadataCategorical']);
+
+    const groupedCellSets = [];
+
+    // from the enabled cell sets keys we get, find their corresponding information
+    groupedTracksKeys
+      .forEach((trackKey) => {
+        const groupBy = allCellSetsGroupBys
+          .find((cellSetGroupBy) => cellSetGroupBy.key === trackKey);
+
+        groupedCellSets.push(groupBy);
+      });
+
+    // About the filtering: If we have failed to find some of the groupbys information,
+    // then ignore those (this is useful for groupbys that sometimes dont show up, like 'samples')
+    return groupedCellSets.filter((groupedCellSet) => groupedCellSet !== undefined);
+  };
+
+  const isInitialRenderRef = useRef(true);
+  const [cellSetsOrder, setCellSetsOrder] = useState(getCellSetsOrder());
+
+  useEffect(() => {
+    if (isInitialRenderRef.current) {
+      isInitialRenderRef.current = false;
+
+      return;
+    }
+
+    if (cellSetsOrder.length === 0) {
+      return;
+    }
+
+    dispatch(
+      updatePlotConfig('interactiveHeatmap', {
```

Comment:
similarly to my comment in the `HeatmapMetadataTrackSettings` component, I suggest that you don't hardcode this here, but rather pass it as props, so that this component is easy to reuse in other parts of the platform.